lib/robots/dor_repo/gis_assembly/generate_structural.rb

```diff
@@ -110,160 +95,96 @@ def file_access_params
               publish: true,
-              sdrPreserve: true,
+              sdrPreserve: preserve,
               shelve: true
             }
-          }
-        end
-
-        def preview_file_params
-          [
-            {
-              type: 'https://cocina.sul.stanford.edu/models/file',
-              externalIdentifier: "https://cocina.sul.stanford.edu/file/#{SecureRandom.uuid}",
-              label: 'preview.jpg',
-              filename: 'preview.jpg',
-              size: preview_objectfile.filesize,
-              version: cocina_object.version,
-              hasMimeType: 'image/jpeg',
-              use: 'master',
-              hasMessageDigests: [
-                {
-                  type: 'sha1',
-                  digest: preview_objectfile.sha1
-                },
-                {
-                  type: 'md5',
-                  digest: preview_objectfile.md5
-                }
-              ],
-              access: file_access_params,
-              administrative: {
-                publish: true,
-                sdrPreserve: true,
-                shelve: true
-              },
-              presentation: preview_presentation_params
-            }
-          ]
+          }.tap do |params|
+            params[:presentation] = presentation if presentation
+          end
         end
 
-        def metadata_file_params
+        DATA_FILE_MIMETYPES =
           [
-            {
-              type: 'https://cocina.sul.stanford.edu/models/file',
-              externalIdentifier: "https://cocina.sul.stanford.edu/file/#{SecureRandom.uuid}",
-              label: esri_metadata_objectfile.filename,
-              filename: esri_metadata_objectfile.filename,
-              size: esri_metadata_objectfile.filesize,
-              version: cocina_object.version,
-              hasMimeType: 'application/xml',
-              use: 'master',
-              hasMessageDigests: [
-                {
-                  type: 'sha1',
-                  digest: esri_metadata_objectfile.sha1
-                },
-                {
-                  type: 'md5',
-                  digest: esri_metadata_objectfile.md5
-                }
-              ],
-              access: file_access_params,
-              administrative: {
-                publish: true,
-                sdrPreserve: true,
-                shelve: true
-              }
-            }
-          ].tap do |params|
-            params.concat(metadata_files)
+            ['.shp', 'application/vnd.shp'],
+            ['.shx', 'application/vnd.shx'],
+            ['.vat.dbf', 'application/octet-stream'],
+            ['.dbf', 'application/vnd.dbf'],
+            ['.prj', 'text/plain'],
+            ['.cpg', 'text/plain'],
+            ['.geojson', 'application/json'],
+            ['.tif', 'image/tiff'],
+            ['.tfw', 'text/plain'],
+            ['.xml', 'application/xml']
+          ].freeze
+
+        def mimetype_for_data_file(objectfile)
```

Comment:
Determines the mimetype, as the mimetype provided by objectfile doesn't match the spec.